src/mem/threadalloc.h

```diff
@@ -137,7 +138,7 @@ namespace snmalloc
     Singleton<pthread_key_t, &pthread_create> p_key;
     // We need to set a non-null value, so that the destructor is called,
     // we never look at the value.
-    pthread_setspecific(p_key.get(), reinterpret_cast<void*>(1));
+    pthread_setspecific(p_key.get(), reinterpret_cast<void*>(p_teardown_val));
```

Comment:
Okay, I prefer your suggestion to make it a static of this function. 

I think you might want to change it to:
```suggestion
    pthread_setspecific(p_key.get(), &p_teardown_val);
```
I think your fix has not addressed the compilers concern just masked it by breaking the analysis pass.

Reply:
in that case we would need to pass the key from a more "meaningful" source as function argument. however no use of this tough.